azkaban-hadoop-security-plugin/src/main/java/azkaban/security/HadoopSecurityManager_H_2_0.java

```diff
@@ -334,12 +334,22 @@ private CredentialProviderWithKeyStore getCustomCredentialProvider(final Props p
 
   private void registerCustomCredential(final Props props, final Credentials hadoopCred,
       final String userToProxy, final Logger jobLogger, final String customCredentialProviderName) {
-    final CredentialProviderWithKeyStore customCredential = getCustomCredentialProvider(
+    final CredentialProvider customCredential = getCustomCredentialProvider(
             props, hadoopCred, jobLogger, customCredentialProviderName);
     final KeyStore keyStore = KeyStoreManager.getInstance().getKeyStore();
     if (keyStore != null) {
       // Containerized Execution.
-      customCredential.setKeyStore(keyStore);
+      try {
+        ((CredentialProviderWithKeyStore)customCredential).setKeyStore(keyStore);
+      } catch (ClassCastException e) {
+        logger.error("Encountered error while casting to "
```

Comment:
Nit: no, this is not the error -- we aren't casting *to* customCredentialProviderName, we are casting customCredentialProviderName to CredentialProviderWithKeyStore. 
But maybe more is less here, and we don't needs to say much at all, because the cause exception `e` will have that info -- see what the message looks like on the ticket.

Reply:
agree, will do.